include/ddc/kernels/fft.hpp

```diff
@@ -581,6 +673,26 @@ void core(
 
 namespace ddc {
 
+/**
+ * @brief Initialize a discrete Fourier space.
+ *
+ * Initialize the (1D) discrete space representing the Fourier discrete dimension associated
+ * to the (1D) spatial mesh passed as argument. It is a N-periodic PeriodicSampling defined between
+ * ka=0 and kb=2*N/(b-a)*pi.
+ *
+ * This value for kb comes from the Nyquist-Shannon theorem: the period of the spectral domain
+ * is kb-ka = 2*pi/cell_size = 2*pi*N/(b-a). The PeriodicSampling then contains cells between coordinates
+ * k=0 and k=2*pi*(N-1)/(b-a), because the cell at coordinate k=2*pi*N/(b-a) is a periodic point (f(ka)=f(kb)).
```

Comment:
Note: `cell_size` is also `(xmax-xmin)/(N-1)`, leading to the upper bound (excluding periodic point) k=2*pi*(N-2)/(xmax-xmin). Both formula are equivalent, this one would avoid relying on the functions `b()` and `a()` (which hide their own formula) which thus could be removed (this is the only place they are called). It would make the formula slightly harder to explain but I think it is better.

Reply:
About the comparison between ddc-fft and numpy.fftfreq. We have `d*n=b-a`.

- `fftfreq` for the `N even` case gives frequencies from `0` to `(N-2)/(b-a)/2`, then (concatenated) from `-N/(b-a)/2` to `-1/(b-a)`, with step `1/(b-a)`.
- `fftfreq` for the `N odd` case gives frequencies from `0` to `(N-1)/(b-a)/2`, then (concatenated) from `-(N-1)/(b-a)/2` to `-1/(b-a)`, with step `1/(b-a)`.
- `fftshift` for the `N even`  case gives frequencies from `-N/(b-a)/2` to `(N-2)/(b-a)/2 with step `1/(b-a)`.
- `fftshift` for the `N odd`  case gives frequencies from `-(N-1)/(b-a)/2` to `(N-1)/(b-a)/2 with step `1/(b-a)`.
- `ddc fft` gives frequencies from `0` to `(N-1)/(b-a)`. But this is a PeriodicSampling, so taking this periodicity in account it exactly corresponds to what is done in `fftfreq`.

Reply:
Here is a MR implementing the a() and b() function, if you prefer it we can merge it first:
https://github.com/CExA-project/ddc/pull/541/files